src/Controls/src/Core/Items/ItemsView.cs

```diff
@@ -139,6 +139,32 @@ public void RemoveLogicalChild(Element element)
 			_logicalChildren.Remove(element);
 			OnChildRemoved(element, oldLogicalIndex);
 			VisualDiagnostics.OnChildRemoved(this, element, oldLogicalIndex);
+			DisconnectLogicalChild(element);
+		}
+
+
+		void DisconnectLogicalChild(Element element)
+		{
+#if ANDROID || IOS || MACCATALYST || WINDOWS
+			if (element.Handler is IPlatformViewHandler platformHandler)
+			{
+				foreach (Element child in element.Descendants())
+				{
+					if (child is VisualElement ve)
+					{
+						ve.Handler?.DisconnectHandler();
+
+						if (ve.Handler is IDisposable veHandlerDisposable)
+							veHandlerDisposable.Dispose();
```

Comment:
If you remove all the `dispose` code does that still work? 

I don't think we want to start pulling dispose patterns in the code here.